velox/exec/Spiller.cpp

```diff
@@ -334,7 +334,7 @@ std::unique_ptr<Spiller::SpillStatus> Spiller::writeSpill(int32_t partition) {
       extractSpillVector(
           run.rows, kTargetBatchRows, kTargetBatchBytes, spillVector, written);
       totalBytes += state_.appendToPartition(partition, spillVector);
-      if (totalBytes > state_.targetFileSize()) {
+      if (needSort() && totalBytes > state_.targetFileSize()) {
```

Comment:
```
if (!needSort() && ...
```
I feel a better way is to pass maximum spill file size in case we do sort for aggregation and order by? Thanks!